test/e2e/internal/notation/registry.go

```diff
@@ -0,0 +1,102 @@
+package notation
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"sync"
+
+	oregistry "oras.land/oras-go/v2/registry"
+)
+
+var (
+	repoId   = 0
+	repoIdMu = sync.Mutex{}
```

Comment:
You may consider [sync/Atomic](https://pkg.go.dev/sync/atomic) instead of a Mutex.

Reply:
Updated to use atomic.AddInt64